cranelift/codegen/src/legalizer/boundary.rs

```diff
@@ -113,12 +114,33 @@ fn legalize_entry_params(func: &mut Function, entry: Block) {
 
         let abi_type = pos.func.signature.params[abi_arg];
         let arg_type = pos.func.dfg.value_type(arg);
+        match abi_type.purpose {
+            ArgumentPurpose::StructArgument(size) => {
```

Comment:
Since this is a match with one arm, `if let ArgumentPurpose::StructArgument(size) = abi_type.purpose {` would look slightly better.